### ISSUE TITLE
100

### ISSUE THREAD

**Issue by vaskonov** [deep-nlp-spring-2020/Task-2-puffofsmoke#1]
**Assignment 2.1**
15/15
1.1: 1/1
1.2: 2/2 Naming constants is a good practice.
1.3: 2/2
1.4: 3/3
1.5: 1/1
1.6: 5/5 `pad_sequence` is also a good choice.
1.7: 1/1

**Assignment 2.2**
2.1: 5/5 Good point to care about weight initialization. Also it is nice to make the class flexible, as it is.
2.2: 1/1
2.3: 1/1
2,4: 3/3
2,5: 3/3 In case you train for quite a lot of epochs it is recommended to implement early stopping based on validation loss, e.g. stop if the model does not improve it for, say, 5 epochs. RNNs are prone to overfitting so you may pass by the optimal model weights.
2.6: 5/5
2.7: 1/1
2.8 5/5
2.9 1/1
2.10: 5/5 Masking with pure `pytorch` is indeed crazy, but you did it, excellent.
2.11: 5/5. Excellent study. 

**Assignment 2.3**
Good point with checking out the labels balance
`LABEL.vocab.freqs`

- Creating the Iterator (2 points)
2 from 2

- Define the RNN-based text classification model (10 points)
10 points from 10

- The training loop (3 points)
3 points from 3
Nice visualization update

- Calculate performance of the trained model (5 points)
5 points from 5
Very nice

- Experiments (10 points)
10 points 10

**Assignment 2.4**

- Creating the Iterator (2 points)
2 points from 2

- Define CNN-based text classification model (8 points)
8 points from 8

- The training loop (3 points)
3 points from 3
Again make sure that the model is not overfitted here

- Calculate performance of the trained model (2 points)
2 points from 2

- Experiments (5 points)
5 points from 5
